### ISSUE TITLE
Feature request: Markdown support (UI and text/markdown from RFC 7763/7764)

### ISSUE THREAD

**Issue by Crissov** [k9mail/k-9#3608]
Related to #1745 (and somewhat to #3171), I wish K9 Mail had an option to send and receive mails that are explicitly labeled as using Markdown (preferably in the Commonmark variant) as specified in [RFC 7763](https://tools.ietf.org/html/rfc7763 "The text/markdown Media Type") and its companion [RFC 7764](https://tools.ietf.org/html/rfc7764 "Guidance on Markdown: Design Philosophies, Stability Strategies, and Select Registrations"). 

Not many mail user agents do implement these yet, unfortunately, and there would need to be some tests whether sending as `text/markdown` *instead of* rather than *in addition to* `text/plain` yields any serious compatibility problems. The expected result is that receiving MUAs just display the Markdown text verbatim. 
Receiving such mails and parsing them for enriched displaying should be implemented regardless.

### Expected behavior

- If K9 had a rich text editor that added formatting to the mail text #1745, there should be an option to use `text/markdown` instead of `text/html` or `text/enriched` ([RFC 1563](https://tools.ietf.org/html/rfc1563)/[1896](https://tools.ietf.org/html/rfc1896)) or `text/richtext` ([RFC 1341](https://tools.ietf.org/html/rfc1341)/[1521](https://tools.ietf.org/html/rfc1521)).
- If K9 detected that the user (supposedly) manually entered Markdown formatting, K9 should offer to send the mail as `text/markdown`.

### Actual behavior

- K9 does not have a rich text editor.
- K9 does not detect the use of formatting characters or codes.
- K9 sends mails as `text/plain` (or `text/html`?).

**Comment by d21d3q** [comment]
How about composing message using markdown and converting to html before sending?
Something like using [Markdown Here](https://markdown-here.com/). In such cases you don't have to worry about other clients, yet you can send rich text messages without rich text editor.

**Comment by yarikoptic** [comment]
Meanwhile today I will be continuing my mutt to do that ;-)

**Comment by fallsdevil** [comment]
Yes, I would also like to be able to write emails in Markdown, at least the simple formatings for now would be great.
